packages/smart-wallet/src/smartWallet.js

```diff
@@ -150,7 +141,7 @@ export const prepareSmartWallet = (baggage, shared) => {
       invitationDisplayInfo: DisplayInfoShape,
       publicMarshaller: M.remotable('Marshaller'),
       zoe: M.eref(M.remotable('ZoeService')),
-      registry: M.remotable('AssetRegistry'),
+      registry: M.remotable('AssetRegistry'), // XXX too strict
```

Comment:
Since this needn't be far, let's drop the Far from `const registry = Far` in WalletFactory and give this a shape. That could be `ReadonlyMapStoreShape`.

Reply:
The registry can't go thru shape validation of any kind as long as it's not passable. And function properties are only allowed on `Far` objects.

This seems to work:

```js
  const { registry: _, ...passableShared } = shared;
  mustMatch(
    harden(passableShared),
    ...
```